src/inference/src/system_conf.cpp

```diff
@@ -38,6 +38,16 @@
 
 namespace ov {
 
+#if defined(OPENVINO_ARCH_ARM64) && defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)
```

Comment:
__ARM_FEATURE_FP16_VECTOR_ARITHMETIC will not be defined, as OpenVINO is not compiled with +fp16 option.